File: Java/compoundlab/src/main/java/com/bbd/abstractFactoryPattern/DuckFactory.java
package com.bbd.abstractFactoryPattern;

public class DuckFactory extends AbstractDuckFactory
{
	public IQuackable createMallardDuck()
	{
		return (new MallardDuck());
	}

	public IQuackable createRedheadDuck()
	{
		return (new RedheadDuck());
	}

	public IQuackable createDuckCall()
	{
		return (new DuckCall());
	}

	public IQuackable createRubberDuck()
	{
		return (new RubberDuck());
	}
}

File: Scala/compoundlab/src/main/scala/DuckFactory.scala
class DuckFactory extends IDuckFactory {
	override def createMallardDuck : IQuackable = {
		new MallardDuck
	}

	override def createRedheadDuck : IQuackable = {
		new RedheadDuck
	}

	override def createDuckCall : IQuackable = {
		new DuckCall
	}

	override def createRubberDuck : IQuackable = {
		new RubberDuck
	}
}
